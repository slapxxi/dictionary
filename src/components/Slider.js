// @flow
import { isEmpty } from 'lodash';
import React, { Component } from 'react';
import glamorous from 'glamorous';

type Props = {
  data: Data,
  index?: number,
  renderSlide: (any) => any,
};

type Data = Array<any>;

class Slider extends Component<Props> {
  render() {
    const { data, renderSlide, index = 0 } = this.props;
    if (isEmpty(data)) {
      return null;
    }
    const item = data[index];
    return <Container>{renderSlide({ item })}</Container>;
  }
}

const Container = glamorous.div({
  display: 'flex',
  flex: 1,
  flexBasis: '100%',
  alignSelf: 'stretch',
});

export default Slider;
