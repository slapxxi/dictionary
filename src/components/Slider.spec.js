import React from 'react';
import { mount } from 'enzyme';
import Slider from './Slider';

const slides = [{ text: 'test' }];

it('renders slides', () => {
  const component = mount(
    <Slider
      data={slides}
      renderSlide={({ item }) => <div id="item">{item.text}</div>}
    />,
  );
  expect(component.find('#item').length).toEqual(1);
});

it('renders nothing if data is empty', () => {
  const component = mount(<Slider data={[]} />);
  expect(component.html()).toBeNull();
});

it('animates transitions between received props', () => {
  // body...
});
