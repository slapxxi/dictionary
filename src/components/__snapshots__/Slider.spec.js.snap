// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders nothing if data is empty 1`] = `null`;

exports[`renders slides 1`] = `
.css-0,
[data-css-0] {
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  flex: 1;
  flex-basis: 100%;
  align-self: stretch;
  -webkit-flex: 1;
  -webkit-flex-basis: 100%;
  -webkit-align-self: stretch;
}

<div
  class="css-0"
>
  <div>
    first
  </div>
  <div>
    first
  </div>
  <div>
    first
  </div>
</div>
`;
